src/main/java/com/encora/Wine.java

```diff
@@ -0,0 +1,62 @@
+package com.encora;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.Comparator;
+
+@AllArgsConstructor
+@Data
+class Wine implements Comparable<Wine> {
+
+    private String name;
+    private String kind;
+    private Double pH;
+    private Double alcohol;
+    private int quality;
+
+        @Override
+        public int compareTo(Wine o) {
+            // Compare by pH and kind
+            return Comparator.comparing(Wine::getPH).compare(this, o);
+        }
+/*
+        @Override
+    public int compareTo(Apple other) {
+        int result = this.name.compareTo(other.name);
+        if (result == 0) {
+            result = this.pH.compareTo(other.pH);
+        }
+        if (result == 0) {
+            result = Integer.compare(this.quality, other.quality);
+        }
+        return result;
+    }
+
+    @Override
+    public int compareTo(Wine other) {
+        if (this.pH < other.pH) {
```

Comment:
Nitpick: I know this is commented code but it can be simplified to 
`return this.pH - other.pH`. 

`Comparable` contract says:
```
Compares this object with the specified object for order. Returns a negative integer, zero, or a positive integer as this object is less than, equal to, or greater than the specified object.
```

so that's a nice trick to avoid these conditions.